### ISSUE TITLE
#255 Stage A0 fresh execution after #262 infrastructure invalidation

### ISSUE THREAD

**Issue by lisbun** [lisbun/lisjong-arena#309]
Refs #255
Refs #259 — COMPLETE / A0 PROTOCOL LOCKED — DOWNSTREAM ENABLED
Refs #262 — COMPLETE / STOP / INVALID
Refs #307 — COMPLETE / FRESH EXECUTION ADMISSIBLE
Refs PR #308

## Purpose

Execute Stage A0 again as a **fresh scientific execution** after #262 was invalidated by an execution-infrastructure fingerprint bug before optimizer training began.

This Issue is not a continuation or reopening of #262. It is a new execution authorization with:
- the exact same final #259 scientific Lock B;
- corrected Arena execution revision;
- a new immutable output root;
- no reuse, deletion, or overwrite of the #262 result artifact.

## Why a fresh execution is admissible

#307 recorded:

```text
#262 terminal result = STOP / INVALID
Gate A0.5 result exposed = false
downstream result exposed = false
protected TEST evaluated = false
failure occurred before optimizer training began
failure arose from execution machinery, not scientific outcome

FRESH EXECUTION ADMISSIBLE
```

The #307/#308 correction changes only initialization identity checking:
- old: SHA-256 of `torch.save(state_dict)` container bytes;
- new: canonical tensor-content identity binding key + dtype + shape + raw CPU tensor bytes;
- A/T initial states are also compared directly by key/dtype/shape/value.

Checkpoint file SHA-256 integrity checks remain unchanged.

## Bound scientific protocol

Use final #259 Lock B unchanged.

Lock B identity:

```text
a7f1c471a8c2f983c5efab9147651eb48b8788eecb1aa296222e21fd9418e93e
```

No scientific parameter from #259/#262 may change.

In particular retain exactly:
- TRAIN 245..264;
- VALIDATION 265..270;
- protected TEST 271..276 unread;
- training seeds 0,1,2;
- anchor seed 0;
- A = Policy only;
- T = Policy + structural-Tenpai auxiliary;
- lambda_tenpai = 1.0;
- no class weighting;
- checkpoint selection by lowest VALIDATION choice-row masked Policy CE only;
- locked Baseline 0 / Baseline 1;
- locked Gate A0.5 metric, blocks, interval, and PASS rule;
- if Gate PASS, exact locked downstream plan: mechanism-riichi-defense x3, ordered seeds 37700..37905, four focal-seat rotations, 206 paired blocks / 1,648 total games;
- no rescue rerun and no result-driven changes.

## Corrected execution revision

Arena:

```text
3a995e89deff3e89eff9e755b4d1f323e96c12e8
```

Locked downstream runtime remains:

```text
lisjong
f29d129c67e5232d06563c6e457754377734ed14

lisjong-engine
8735e89e1aea000ab59368d0368d476787827741

RiichiEnv
0.4.10

PyTorch
2.13.0
```

Use the operator's Python 3.14.6 environment unless an explicit preflight mismatch requires STOP / INVALID.

## Phase X0 — non-scientific runtime regression check

Before touching scientific artifacts:

1. create a clean worktree at the corrected Arena revision;
2. create a fresh dedicated venv;
3. install the Arena ML/dev dependencies;
4. force-install the locked downstream lisjong revision;
5. verify execution provenance;
6. run the focused Stage A0 initialization regression test under the actual operator runtime.

The regression must establish that:
- strict-loaded Policy state is key/dtype/shape/value-identical;
- canonical semantic fingerprints match;
- the real `train_seed_pair()` initialization boundary succeeds.

This phase must not read TRAIN/VALIDATION/TEST artifacts.

If this non-scientific regression fails:

```text
DO NOT START SCIENTIFIC EXECUTION
```

A failure here does not authorize changing the scientific protocol.

## Phase X1 — scientific preflight

Only after X0 passes:

- strict-read exact final #259 Lock B;
- verify all locked scientific artifact identities and runtime provenance;
- verify protected TEST remains unread;
- verify new output root does not already exist.

Use exactly this new output root:

```text
C:\Dev\lisjong-artifacts\issue-309-stage-a0-execution
```

Never delete or overwrite the #262 artifact:

```text
C:\Dev\lisjong-artifacts\issue-262-stage-a0-execution
```

## Phase X2 — one-shot execution

Run the corrected Stage A0 executor exactly once.

Required routing remains the #262/#259 routing:

```text
PREFLIGHT PASS
  -> train A/T for all three locked seeds
  -> freeze all six checkpoints
  -> Gate A0.5

Gate FAIL
  -> NO DENSE AUXILIARY LEARNING SIGNAL
  -> no downstream

Gate PASS
  -> execute only the already-locked downstream plan
```

Any scientific runtime/execution failure after X1 starts follows the existing locked fail-closed rule:

```text
STOP / INVALID
```

No retry is authorized inside this Issue unless an explicit pre-existing locked technical-retry rule applies.

## Result exposure boundary

After first Gate A0.5 result exposure:
- no retraining;
- no seed replacement;
- no lambda/baseline/checkpoint change;
- no extra VALIDATION data.

After first downstream result exposure:
- no extra seeds;
- no alternate opponents;
- no second confirmation run.

## Acceptance criteria

- [ ] Corrected Arena revision is exact.
- [ ] Locked lisjong / lisjong-engine / RiichiEnv / PyTorch identities match.
- [ ] Non-scientific initialization regression passes before scientific data access.
- [ ] Exact #259 Lock B strict-read succeeds.
- [ ] #262 artifact is preserved unchanged.
- [ ] A new output root is used.
- [ ] Protected TEST remains unread.
- [ ] Exactly seeds 0,1,2 are used.
- [ ] A/T initialization is content-identical within each seed.
- [ ] Six checkpoints are frozen before Gate execution.
- [ ] Gate uses exact locked Baseline 1 / metric / interval / PASS rule.
- [ ] Downstream runs only on Gate PASS.
- [ ] If run, downstream uses exact locked opponents/seeds/rotations/game budget.
- [ ] Completion artifact strict-read succeeds.
- [ ] Exactly one valid final outcome is recorded.
- [ ] No rescue rerun occurs.

## Valid final outcomes

Exactly one:

```text
NO DENSE AUXILIARY LEARNING SIGNAL

TENPAI REPRESENTATION SIGNAL / POLICY VALUE NOT TESTED DUE TO POWER

TENPAI REPRESENTATION SIGNAL / POLICY VALUE NOT ESTABLISHED

TENPAI SUPERVISION VALUE SIGNAL

STOP / INVALID
```

**Comment by lisbun** [comment]
## X0 operator-local handoff

The corrected execution revision and focused regression coverage have been rechecked.

Before any scientific artifact access, execute X0 in a fresh worktree:

```powershell
cd C:\Dev\lisjong-arena
git fetch origin

git worktree add C:\Dev\lisjong-arena-309 `
  3a995e89deff3e89eff9e755b4d1f323e96c12e8

cd C:\Dev\lisjong-arena-309

py -3.14 -m venv .venv
.\.venv\Scripts\Activate.ps1

python -m pip install -e ".[ml,dev]"

python -m pip install --no-deps --force-reinstall `
  "lisjong @ git+https://github.com/lisbun/lisjong.git@f29d129c67e5232d06563c6e457754377734ed14"

git rev-parse HEAD
git status --short

python -c "import json, torch; from lisjong_arena.single_round_artifact import collect_execution_provenance, execution_provenance_to_dict; print(torch.__version__); print(json.dumps(execution_provenance_to_dict(collect_execution_provenance()), indent=2))"

python -m unittest discover -s tests -p "test_stage_a0_tenpai_execution_ml.py" -v
```

Expected identities:
- Arena `3a995e89deff3e89eff9e755b4d1f323e96c12e8`
- lisjong `f29d129c67e5232d06563c6e457754377734ed14`
- lisjong-engine `8735e89e1aea000ab59368d0368d476787827741`
- RiichiEnv `0.4.10`
- PyTorch `2.13.0`
- operator Python `3.14.6`
- clean `git status --short`

The focused test file uses synthetic/fake fixtures only; it does not read #259/#262 scientific TRAIN, VALIDATION, or protected TEST artifacts.

Do not start the scientific `execute` command until X0 is green.

**Comment by lisbun** [comment]
## X0 PASS — operator-local regression

Operator-local X0 completed successfully on Windows / Python 3.14.6.

Verified:
- Arena: `3a995e89deff3e89eff9e755b4d1f323e96c12e8`
- lisjong: `f29d129c67e5232d06563c6e457754377734ed14`
- lisjong-engine: `8735e89e1aea000ab59368d0368d476787827741`
- RiichiEnv: `0.4.10`
- PyTorch: `2.13.0+cpu`
- Python: `3.14.6`
- worktree clean

Focused ML regression result:

```text
Ran 4 tests in 39.878s
OK
```

This includes:
- semantic fingerprint binds dtype/shape/raw tensor bytes;
- semantic fingerprint survives strict load while legacy torch.save serialization hash can drift;
- checkpoint strict load/serving;
- real train_seed_pair() initialization boundary.

No scientific TRAIN/VALIDATION/TEST artifacts were read during X0.

X0 classification:

```text
PASS
```

Proceeding to X1 artifact/preflight checks only. Scientific execute has not started yet.

**Comment by lisbun** [comment]
## Final scientific result — Stage A0

Fresh execution under #309 completed successfully under the unchanged #259 Lock B.

Execution identities:
- Arena `3a995e89deff3e89eff9e755b4d1f323e96c12e8`
- lisjong `f29d129c67e5232d06563c6e457754377734ed14`
- lisjong-engine `8735e89e1aea000ab59368d0368d476787827741`
- RiichiEnv `0.4.10`
- Python `3.14.6`
- PyTorch `2.13.0+cpu`

Preflight:
```text
PREFLIGHT PASS
scientific rows: TRAIN=9116 VALIDATION=2555
```

All six locked checkpoints were produced and frozen before Gate A0.5:

```text
A_seed0  03428acdbd14503e1458ac9f34ab6690b77ae2565d1421db90d60e172912b516
T_seed0  47584a6c415bceb811bcc34c27fd3f4719b04b0e6e4aa361c4a472125a547127
A_seed1  21df349d1ba7b4cc3587d907077af5b725e183466306c1d34325b64461f330c5
T_seed1  b9bc3b012968d44100349bfc201662d57ab132d2e138cae9dbcaaf5c29d7906a
A_seed2  fd9d1fbd714af23a14bb1284cc988f7fa974dc57193f8f9ebb14f1635736bcdf
T_seed2  338ebea7646c3e6f318430092949bc95bd4eb921e82169622fa0ce9e13e66ff3
```

Gate result:

```text
TENPAI LEARNABILITY FAIL
```

Per the locked route, downstream A-vs-T game evaluation was **not run**.

Final completion artifact:

```text
result_identity =
96c54871894c8269d730c29213f774a0cb36800fd55eb6902deeaef7b172f587

primary_outcome =
NO DENSE AUXILIARY LEARNING SIGNAL

protected_test_evaluated = false
```

The CLI strict-read the completion artifact before printing the final result, so completion-artifact readback succeeded.

This is a valid scientific terminal result, not STOP / INVALID.

Routing:
```text
Stage A0 = FAIL
downstream = NOT RUN
Stage A1 = NOT ELIGIBLE
parent #255 = REFORMULATE
```

No rescue run is authorized.
